standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5484,6 +5509,29 @@ private void fireReadTablePreEvent(String catName, String dbName, String tblName
     }
   }
 
+  /**
+   * Fire a pre-event for read database operation, if there are any
+   * pre-event listeners registered
+   */
+  private void fireReadDatabasePreEvent(final String name)
+          throws MetaException, NoSuchObjectException {
+    if(preListeners.size() > 0) {
+      // do this only if there is a pre event listener registered (avoid unnecessary
+      // metastore api call)
+      String[] parsedDbName = parseDbName(name, conf);
+      Database db = null;
+      try {
+        db = get_database_core(parsedDbName[CAT_NAME], parsedDbName[DB_NAME]);
+        if (db == null) {
+          throw new NoSuchObjectException("Database: " + name + " not found");
+        }
+      } catch(MetaException | NoSuchObjectException e) {
+        throw new RuntimeException(e);
```

Comment:
you are packaging all exceptions (including `MetaException`) into a `RuntimeException` 
a) is this really neccessary?
b) at the call site the catch is only about `MetaException` which in its current form will never be thrown

does this work correctly?
could you add a test?

note: I think it might worth to consider to flatten this code into the new method I suggested to create in the other comment

Reply:
firePreEvent throws metaexception when the we don't have permission on the database hence i have catched metaexception and checked for that,

yes it works. I have verified this on the cluster. I am not sure how to add tests for the same. Let me explore that.

Reply:
firePreEvent throws metaexception when the we don't have permission on the database hence i have catched metaexception and checked for that,

yes it works. I have verified this on the cluster. I am not sure how to add tests for the same. Let me explore that.

Reply:
Added tests